src/main/java/com/smartcar/sdk/AuthClient.java

```diff
@@ -310,6 +312,101 @@ public AuthVehicleInfo build() {
     }
   }
 
+  /**
+   * A class that creates a custom AuthUrlConfig object, which is used
+   * when generating an authentication URL.
+   */
+  public static class AuthUrlConfig {
+    private boolean forcePrompt;
+    private String make;
+    private boolean singleSelect;
+    private String state;
+    
+    /** 
+     * A function that creates a map with query params and their values.
+     */
+    public HashMap getParams() {
+      HashMap<String, String> list = new HashMap<>();
+
+      list.put("single_select", this.singleSelect ? "true" : "false");
```

Comment:
I wonder what the benefit of doing this is over 
```
if (this.singleSelect) {
  list.put("single_select", "true");
}
```
I guess if we are not relying on the default behavior, but maybe we should?

Also, what about an alternative, not sure how this will work with the query string.

```
list.put("single_select", this.singleSelect);
```

Reply:
I like the idea of relying on default behavior. It needs to be a `string` because that's what the `addQueryParameter` expects